compiler/rustc_const_eval/src/interpret/validity.rs

```diff
@@ -594,7 +594,10 @@ impl<'rt, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> ValidityVisitor<'rt, 'mir, '
                 Ok(true)
             }
             ty::Adt(def, ..) if def.is_box() => {
-                self.check_safe_pointer(value, "box")?;
+                let unique = self.ecx.operand_field(value, 0)?;
+                let nonnull = self.ecx.operand_field(&unique, 0)?;
+                let ptr = self.ecx.operand_field(&nonnull, 0)?;
```

Comment:
This will fail to validate all the other fields we might be skipping here.

In particular, if the allocator is, e.g., `bool`, then we need to check that it is a valid bool.

Reply:
It validates the allocator now.